controllers/storageclient_controller.go

```diff
@@ -314,65 +278,65 @@ func (s *StorageClientReconciler) onboardConsumer(instance *v1alpha1.StorageClie
 	}
 
-	instance.Status.ConsumerID = response.StorageConsumerUUID
-	instance.Status.Phase = v1alpha1.StorageClientOnboarding
+	r.storageClient.Status.ConsumerID = response.StorageConsumerUUID
+	r.storageClient.Status.Phase = v1alpha1.StorageClientOnboarding
 
-	s.Log.Info("onboarding started")
+	r.Log.Info("onboarding started")
 	return reconcile.Result{Requeue: true}, nil
 }
 
-func (s *StorageClientReconciler) acknowledgeOnboarding(instance *v1alpha1.StorageClient, externalClusterClient *providerClient.OCSProviderClient) (reconcile.Result, error) {
+func (r *StorageClientReconciler) acknowledgeOnboarding(externalClusterClient *providerClient.OCSProviderClient) (reconcile.Result, error) {
 
-	_, err := externalClusterClient.AcknowledgeOnboarding(s.ctx, instance.Status.ConsumerID)
+	_, err := externalClusterClient.AcknowledgeOnboarding(r.ctx, r.storageClient.Status.ConsumerID)
 	if err != nil {
 		if st, ok := status.FromError(err); ok {
-			s.logGrpcErrorAndReportEvent(instance, AcknowledgeOnboarding, err, st.Code())
+			r.logGrpcErrorAndReportEvent(AcknowledgeOnboarding, err, st.Code())
 		}
-		s.Log.Error(err, "Failed to acknowledge onboarding.")
+		r.Log.Error(err, "Failed to acknowledge onboarding.")
 		return reconcile.Result{}, fmt.Errorf("failed to acknowledge onboarding: %v", err)
 	}
-	instance.Status.Phase = v1alpha1.StorageClientConnected
+	r.storageClient.Status.Phase = v1alpha1.StorageClientConnected
 
-	s.Log.Info("Onboarding is acknowledged successfully.")
+	r.Log.Info("Onboarding is acknowledged successfully.")
 	return reconcile.Result{Requeue: true}, nil
 }
 
 // offboardConsumer makes an API call to the external storage provider cluster for offboarding
-func (s *StorageClientReconciler) offboardConsumer(instance *v1alpha1.StorageClient, externalClusterClient *providerClient.OCSProviderClient) (reconcile.Result, error) {
+func (r *StorageClientReconciler) offboardConsumer(externalClusterClient *providerClient.OCSProviderClient) (reconcile.Result, error) {
 
-	_, err := externalClusterClient.OffboardConsumer(s.ctx, instance.Status.ConsumerID)
+	_, err := externalClusterClient.OffboardConsumer(r.ctx, r.storageClient.Status.ConsumerID)
 	if err != nil {
 		if st, ok := status.FromError(err); ok {
-			s.logGrpcErrorAndReportEvent(instance, OffboardConsumer, err, st.Code())
+			r.logGrpcErrorAndReportEvent(OffboardConsumer, err, st.Code())
 		}
 		return reconcile.Result{}, fmt.Errorf("failed to offboard consumer: %v", err)
 	}
 
 	return reconcile.Result{}, nil
 }
 
-func (s *StorageClientReconciler) verifyNoStorageClaimsExist(instance *v1alpha1.StorageClient) error {
+func (r *StorageClientReconciler) verifyNoStorageClaimsExist() error {
 
 	storageClaims := &v1alpha1.StorageClaimList{}
-	err := s.Client.List(s.ctx,
+	err := r.Client.List(r.ctx,
 		storageClaims,
-		client.MatchingFields{storageClientAnnotationIndexName: client.ObjectKeyFromObject(instance).String()},
+		client.MatchingFields{ownerIndexName: r.storageClient.Name},
```

Comment:
You should track by owner UID, not owner name. OwnerRefernace might be stale and pointing to a previous storage client that had a similar name.

Reply:
Yes, actual bug in my code!

Reply:
done.